plugins/activity/activity_pool.go

```diff
@@ -128,6 +136,10 @@ func (pool *activityPool) executeActivity(ctx context.Context, input rrt.RRPaylo
 		}
 	)
 
+	/*
+		TODO: next iteration should avoid processing payloads anyhow, the processing should be done on PHP end.
+	*/
+
 	// todo: optimize
 	for _, value := range input.Data {
```

Comment:
For a little optimization you may use`for` loop instead of range, since the second `arg`  in range is copy unnecessary copy on `input.Data`. Or use form of
```go
for key := range input.Data {
vData, err := jsoniter.Marshal(input.Data[key])
out.Data = append(out.Data, input.Data[key])
```